### ISSUE TITLE
Traceback when domain already exists

### ISSUE THREAD

**Issue by tripleee** [Charcoal-SE/metasmoke#881]
https://metasmoke.erwaysoftware.com/domains/new gives me a form which however gives me a traceback and a redirect to https://metasmoke.erwaysoftware.com/domains/no_post_create when I try to submit it.

The domain I wanted to add was `drsumannag.in` and while I understand that I can't create a duplicate, an error message would be more appropriate and user-friendly than a traceback.

> ## ActionController::UrlGenerationError in SpamDomainsController#create
> ### No route matches {:action=&gt;"show", :controller=&gt;"spam_domains", :id=&gt;nil}
> Extracted source (around line #53):
>
> 51 @domain.save
> 52
> 53      redirect_to action: :show, id: @domain.id <<<<<<
> 54   end
> 55
> 56    def update
>
>`Rails.root: /var/railsapps/metasmoke/releases/20210909152825
> 
>  ``app/controllers/spam_domains_controller.rb:53:in `create'``
>
> ## Request
>
> Parameters:
>
> {"utf8"=&gt;"✓", "authenticity_token"=&gt;"soooseeeekret==", "spam_domain"=&gt;{"domain"=&gt;"drsumannag.in"}}

**Comment by makyen** [comment]
I was working on the email version of your report, so the version saying you couldn't create the domain. I tried creating that domain and it worked fine when I tried [creating that domain](https://metasmoke.erwaysoftware.com/domains/84716). I've created a couple/few domains in the last few/several days and experienced no problems. That I created the domain may be the reason you're now getting a different error.

Now that the domain exists, I, also, get a traceback.
